test/e2e/tests/change-language.spec.js

```diff
@@ -0,0 +1,325 @@
+          By.css('.settings-page__content-label'),
+        );
+        assert.equal(
+          await labelText.getText(),
+          'Current Language',
+          'Language did not change',
+        );
+      },
+    );
+  });
+
+  it('User selects "Español" and verify that language persists with page refresh and sessions', async function () {
+    if (process.env.MULTICHAIN) {
+      return;
+    }
+
+    const languageIndex = 10;
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+        failOnConsoleError: false,
+      },
+
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await changeLanguage({ driver, languageIndex });
+
+        await driver.refresh();
+        const advanceText = await driver.findElement({
+          tag: 'span',
+          text: 'Idioma actual',
+        });
+        assert.equal(
+          await advanceText.getText(),
+          'Idioma Actual',
+          'Language did not change',
+        );
+
+        await driver.navigate();
+
+        await driver.delay(2000);
+        // Validating the tooltip
+        assert.equal(
+          await driver.isElementPresent(
+            '[data-original-title="Copiar al Portapapeles"]',
+          ),
+          true,
+          'Language changes is not reflected on the toolTip',
+        );
+      },
+    );
+  });
+
+  it('User selects "Dansk" language and verify that navigating to a different page does not affect', async function () {
+    const languageIndex = 6;
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+        failOnConsoleError: false,
+      },
+
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await changeLanguage({ driver, languageIndex });
+
+        await driver.clickElement({ text: 'Avanceret', tag: 'div' });
+
+        // Confirm that the language change is reflected in search box water text
+        assert.equal(
+          await driver.isElementPresent('[placeholder="Søg"]'),
+          true,
+          'Water text in the search box does not match with the selected language',
+        );
+
+        // Confirm that the language change is reflected in headers
+
+        assert.equal(
+          await driver.isElementPresent({
+            text: 'Indstillinger',
+            tag: 'h3',
+          }),
+          true,
+          'Language change is not reflected in headers',
+        );
+
+        // Confirm that the language change is reflected in button
+        const buttonText = await driver.findElement(
+          '[data-testid="auto-lockout-button"]',
+        );
+
+        assert.equal(
+          await buttonText.getText(),
+          'Gem',
+          'Language change is not reflected in button',
+        );
```

Comment:
This is an example of the "find then assert" race condition
https://consensys.slack.com/archives/GTQAGKY5V/p1686238528522549

Reply:
I made the code changes and marking resolved.